lib/test/reporter/spec.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const {
+  ArrayPrototypeJoin,
+  ArrayPrototypePop,
+  ArrayPrototypeShift,
+  ArrayPrototypeUnshift,
+  RegExpPrototypeSymbolSplit,
+  SafeMap,
+  StringPrototypeRepeat,
+} = primordials;
+const assert = require('assert');
+const Transform = require('internal/streams/transform');
+const { inspectWithNoCustomRetry } = require('internal/errors');
+const { green, blue, red, white, gray } = require('internal/util/colors');
+
+
+const inspectOptions = { __proto__: null, colors: true, breakLength: Infinity };
+
+const colors = {
+  '__proto__': null,
+  'test:fail': red,
+  'test:pass': green,
+  'test:diagnostic': blue,
+};
+const symbols = {
+  '__proto__': null,
+  'test:fail': '\u2716 ',
+  'test:pass': '\u2714 ',
+  'test:diagnostic': '\u2139 ',
+  'arrow:right': '\u25B6 ',
+};
+class SpecReporter extends Transform {
+  #stack = [];
+  #reported = [];
+  #indentMemo = new SafeMap();
+
+  constructor() {
+    super({ writableObjectMode: true });
+  }
+
+  #indent(nesting) {
+    let value = this.#indentMemo.get(nesting);
+    if (value === undefined) {
+      value = StringPrototypeRepeat('  ', nesting);
+      this.#indentMemo.set(nesting, value);
+    }
+
+    return value;
+  }
+  #formatError(error, indent) {
+    if (!error) return '';
+    const err = error.code === 'ERR_TEST_FAILURE' ? error.cause : error;
+    const message = ArrayPrototypeJoin(
+      RegExpPrototypeSymbolSplit(
+        /\r?\n/,
```

Comment:
It's available in `primordials`.
```suggestion
        hardenRegExp(/\r?\n/),
```